src/8a160_len_700.c

```diff
@@ -18,7 +18,32 @@ typedef struct {
 
 extern struct_func_800F0CB0 D_8010D000[];
 
-INCLUDE_ASM(s32, "8a160_len_700", func_800F0CB0);
+void func_800F0CB0(s32 arg0, f32 arg1, f32 arg2, f32 arg3) {
+    s32 temp_v1;
+    struct_func_800F0CB0* temp_a1;
+    s32 phi_v1;
+    sub_struct_func_800F0CB0* phi_a1;
+
+    temp_a1 = &D_8010D000[arg0];
```

Comment:
We typically like to arrange variable declarations at the beginning of the function so that we can define them and declare them on the same line. So the beginning of the function would ideally look like:

```
struct_func_800F0CB0* temp_a1 = &D_8010D000[arg0];
sub_struct_func_800F0CB0* phi_a1 = &temp_a1_unk00[0];
s32 temp_v1 = 0x39;
s32 phi_v1;
```